packages/admin-ui/src/Select/SelectPresenter.ts

```diff
@@ -0,0 +1,75 @@
+import { makeAutoObservable } from "mobx";
+import omit from "lodash/omit";
+import { SelectOptionsVm, SelectProps, SelectTriggerVm, SelectVm } from "./Select";
+import { SelectOption } from "./SelectOption";
+import { SelectOptionMapper } from "~/Select/SelectOptionMapper";
+
+interface ISelectPresenter<TProps extends SelectProps = SelectProps> {
+    vm: {
+        selectVm: SelectVm;
+        selectTriggerVm: SelectTriggerVm;
+        selectOptionsVm: SelectOptionsVm;
+    };
+    init: (props: TProps) => void;
+    changeValue: (value: string) => void;
+    resetValue: () => void;
+}
+
+class SelectPresenter implements ISelectPresenter {
```

Comment:
I've been actually reading a small bit... And I'm a bit confused when it comes to terminology. Kinda have a feeling we're mixing terms maybe? 🤔 

Like... we are using the word Presenter, so that tells me we're using  the [model-view-presenter  pattern](https://en.wikipedia.org/wiki/Model%E2%80%93view%E2%80%93presenter). But then again, we're also using view model, which now brings me to the [MVVM](https://en.wikipedia.org/wiki/Model%E2%80%93view%E2%80%93viewmodel).

I know all of these patterns are similar / basically the same at some point.. But still, did you maybe notice the same?

Reply:
Thank you for pointing that out! 🙌
You’re totally right: there’s a mix of MVP and MVVM going on here, and it could get confusing.

For now, I’m considering leaving the code and experimenting with it more before deciding whether to refactor to stick with MVP or fully embrace MVVM. It feels like an excellent opportunity to test things and see what works best in practice.